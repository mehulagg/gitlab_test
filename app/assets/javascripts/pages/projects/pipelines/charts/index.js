import $ from 'jquery';
import Chart from 'chart.js';

import { lineChartOptions } from '~/lib/utils/chart_utils';

import initProjectPipelinesChartsApp from '~/projects/pipelines/charts/index';

const SUCCESS_LINE_COLOR = '#1aaa55';

const TOTAL_LINE_COLOR = '#707070';

const buildChart = (chartScope, shouldAdjustFontSize) => {
  const data = {
    labels: chartScope.labels,
    datasets: [
      {
        backgroundColor: SUCCESS_LINE_COLOR,
        borderColor: SUCCESS_LINE_COLOR,
        pointBackgroundColor: SUCCESS_LINE_COLOR,
        pointBorderColor: '#fff',
        data: chartScope.successValues,
        fill: 'origin',
      },
      {
        backgroundColor: TOTAL_LINE_COLOR,
        borderColor: TOTAL_LINE_COLOR,
        pointBackgroundColor: TOTAL_LINE_COLOR,
        pointBorderColor: '#EEE',
        data: chartScope.totalValues,
        fill: '-1',
      },
    ],
  };
  const ctx = $(`#${chartScope.scope}Chart`)
    .get(0)
    .getContext('2d');

  return new Chart(ctx, {
    type: 'line',
    data,
    options: lineChartOptions({
      width: ctx.canvas.width,
      numberOfPoints: chartScope.totalValues.length,
      shouldAdjustFontSize,
    }),
  });
};

document.addEventListener('DOMContentLoaded', () => {
  const chartsData = JSON.parse(document.getElementById('pipelinesChartsData').innerHTML);

  // Scale fonts if window width lower than 768px (iPad portrait)
  const shouldAdjustFontSize = window.innerWidth < 768;

  chartsData.forEach(scope => buildChart(scope, shouldAdjustFontSize));
});

document.addEventListener('DOMContentLoaded', initProjectPipelinesChartsApp);
