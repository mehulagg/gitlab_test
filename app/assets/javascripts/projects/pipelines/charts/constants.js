export const CHART_CONTAINER_HEIGHT = 300;

export const INNER_CHART_HEIGHT = 200;

export const X_AXIS_LABEL_ROTATION = 45;

export const X_AXIS_TITLE_OFFSET = 60;
