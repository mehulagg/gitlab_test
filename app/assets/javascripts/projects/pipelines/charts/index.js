import Vue from 'vue';
import ProjectPipelinesCharts from './components/app.vue';

export default () => {
  const el = document.querySelector('#js-project-pipelines-charts-app');
  const {
    countsFailed,
    countsSuccess,
    countsTotal,
    successRatio,
    timesChartLabels,
    timesChartValues,
  } = el.dataset;

  return new Vue({
    el,
    name: 'ProjectPipelinesChartsApp',
    components: {
      ProjectPipelinesCharts,
    },
    render: createElement =>
      createElement(ProjectPipelinesCharts, {
        props: {
          counts: {
            failed: countsFailed,
            success: countsSuccess,
            total: countsTotal,
            successRatio,
          },
          timesChartData: {
            labels: JSON.parse(timesChartLabels),
            values: JSON.parse(timesChartValues),
          },
        },
      }),
  });
};
