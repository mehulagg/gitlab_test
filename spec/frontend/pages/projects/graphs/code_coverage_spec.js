import { shallowMount } from '@vue/test-utils';
import { GlDropdown, GlDropdownItem } from '@gitlab/ui';
import { GlAreaChart } from '@gitlab/ui/dist/charts';
import CodeCoverage from '~/pages/projects/graphs/components/CodeCoverage.vue';

describe('Code Coverage', () => {
  let wrapper;

  const props = {
    languages: [
      {
        color: '#2c3e50',
        highlight: '#2c3e50',
        label: 'Vue',
        value: 71.62,
      },
      {
        color: '#e34c26',
        highlight: '#e34c26',
        label: 'HTML',
        value: 17.33,
      },
      {
        color: '#f1e05a',
        highlight: '#f1e05a',
        label: 'JavaScript',
        value: 11.05,
      },
    ],
  };

  const createComponent = (props) => {
    wrapper = shallowMount(CodeCoverage, {
      propsData: {
        ...props,
      },
    });
  };

  beforeEach(() => {
    createComponent(props);
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders the dropdown with all languages', () => {
    expect(wrapper.contains(GlDropdown)).toBeDefined();
    expect(wrapper.findAll(GlDropdownItem)).toHaveLength(3);
  });

  it('renders the area chart', () => {
    expect(wrapper.contains(GlAreaChart)).toBeDefined();
  });
  
  it('matches the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

});
