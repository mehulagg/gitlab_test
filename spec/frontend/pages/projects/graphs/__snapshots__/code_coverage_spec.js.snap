// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Coverage matches the snapshot 1`] = `
<div>
  <div
    class="d-flex justify-content-between align-items-center"
  >
    <h4
      class="sub-header"
    >
      Code coverage statistics for master {range-date}
    </h4>
     
    <button
      class="btn btn-light h-100 pl-2 pr-2 pt-0 pb-0"
      style="font-size: 11px;"
    >
      
      Download raw data (.csv)
    
    </button>
  </div>
   
  <div
    class="d-flex justify-content-end mr-5 mt-3"
  >
    <gl-dropdown-stub
      text="Vue"
    >
      <gl-dropdown-item-stub>
        <div
          class="d-flex"
        >
          <gl-icon-stub
            class="position-absolute"
            name="mobile-issue-close"
            size="16"
          />
           
          <span
            class="d-flex align-items-center ml-4"
          >
            <span
              class="mr-2"
              style="color: rgb(44, 62, 80);"
            >
              <gl-icon-stub
                name="stop"
                size="16"
              />
            </span>
            Vue
            
          </span>
        </div>
      </gl-dropdown-item-stub>
      <gl-dropdown-item-stub>
        <div
          class="d-flex"
        >
          <!---->
           
          <span
            class="d-flex align-items-center ml-4"
          >
            <span
              class="mr-2"
              style="color: rgb(227, 76, 38);"
            >
              <gl-icon-stub
                name="stop"
                size="16"
              />
            </span>
            HTML
            
          </span>
        </div>
      </gl-dropdown-item-stub>
      <gl-dropdown-item-stub>
        <div
          class="d-flex"
        >
          <!---->
           
          <span
            class="d-flex align-items-center ml-4"
          >
            <span
              class="mr-2"
              style="color: rgb(241, 224, 90);"
            >
              <gl-icon-stub
                name="stop"
                size="16"
              />
            </span>
            JavaScript
            
          </span>
        </div>
      </gl-dropdown-item-stub>
    </gl-dropdown-stub>
  </div>
   
  <gl-area-chart-stub
    data="[object Object]"
    height="250"
    includelegendavgmax="true"
    legendaveragetext="Avg"
    legendmaxtext="Max"
    option="[object Object]"
    thresholds=""
  />
</div>
`;
